crypto/evp/evp_enc.c

```diff
@@ -344,24 +344,27 @@ static int evp_cipher_init_internal(EVP_CIPHER_CTX *ctx,
             n = EVP_CIPHER_CTX_get_iv_length(ctx);
             if (!ossl_assert(n >= 0 && n <= (int)sizeof(ctx->iv)))
                     return 0;
-            if (iv)
+            if (iv != NULL)
                 memcpy(ctx->oiv, iv, EVP_CIPHER_CTX_get_iv_length(ctx));
             memcpy(ctx->iv, ctx->oiv, EVP_CIPHER_CTX_get_iv_length(ctx));
```

Comment:
why not use `n` instead of `EVP_CIPHER_CTX_get_iv_length(ctx)` here?

Reply:
Pre-existing code I didn't look at....

I've made the change.